### PR TITLE
[scaffolder-react] Add an overridable className to template stepper

### PR DESCRIPTION
**Pull request by nikolarAutodesk**
#### :heavy_check_mark: Checklist
Our design team has mentioned this to us developers on many instances:
"[There is research to support that left alignment is better](https://adamsilver.io/blog/where-to-put-buttons-on-forms/)" 
Though there are many opportunities to improve this through out, we saw Templates as an easy win 🤞 

Adding an overridable className `BackstageTemplateStepper`, so that teams can update the template stepper styling like this if they want to:
Screenshot 2024-10-03 at 2 24 12 PM

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by Rugvip — CHANGES_REQUESTED** [review]
Sorry this is mostly down to opinion and consistency, and we generally try to keep action buttons to the right.

**Comment by Rugvip** [comment]
See "Placement" at https://m2.material.io/components/buttons#text-button as well

**Comment by nikolarAutodesk** [comment]
Fair enough @Rugvip! What are your thoughts on just adding an overridable class? I'm hoping this is a good compromise?

**Comment by benjdlambert** [comment]
@nikolarAutodesk think that this is a better approach, but you'll need to update the `api-reports` using `yarn build:api-reports` and then commit and push the changes :tada:

**Comment by vinzscam** [comment]
Closing and re-opening as DCO check went bananas 😅

File: plugins/scaffolder-react/src/next/components/Stepper/Stepper.tsx [hunk 1]
```diff
@@ -56,20 +56,28 @@ import { merge } from 'lodash';
...
-    display: 'flex',
-    flexDirection: 'row',
-    justifyContent: 'right',
-    marginTop: theme.spacing(2),
-  },
-  formWrapper: {
-    padding: theme.spacing(2),
-  },
-}));
+export type BackstageTemplateStepperClassKey =
```
[the remark this answers is not in the document]
Comment (nikolarAutodesk) [reply plugins/scaffolder-react/src/next/components/Stepper/Stepper.tsx:60]: thank you! That seemed to work

**Review by benjdlambert — APPROVED** [review]
Nice thanks :tada:

**Pull request merged**